types/yarnpkg__lockfile/index.d.ts

```diff
@@ -1,32 +1,31 @@
 // Type definitions for @yarnpkg/lockfile 1.1
 // Project: https://github.com/yarnpkg/yarn/tree/master/packages/lockfile
 // Definitions by: Eric Wang <https://github.com/fa93hws>
+//                 rigwild <https://github.com/rigwild>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 export interface Dependency {
-  [packageName: string]: string;
+    [packageName: string]: string;
 }
 
 export interface FirstLevelDependency {
-  version: string;
-  resolved?: string;
-  dependencies?: Dependency;
+    version: string;
+    resolved?: string;
+    integrity?: string;
+    dependencies?: Dependency;
 }
 
 export interface LockFileObject {
-  [packageName: string]: FirstLevelDependency;
+    [packageName: string]: FirstLevelDependency;
 }
 
 export function parse(
-  file: string,
-  fileLoc?: string,
-): {
-  type: 'success' | 'merge' | 'conflict';
-  object: any;
-};
+    file: string,
+    fileLoc?: string
+): { type: 'success' | 'merge'; object: LockFileObject } | { type: 'conflict'; object: {} };
 
 export function stringify(
-  json: any,
-  noHeader?: boolean,
-  enableVersions?: boolean,
+    json: LockFileObject,
```

Comment:
There are a few comments around this in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/37756

Reply:
tbh, I agree with you that it's better to set typings and ignore very edge cases like mentionned. In such case (and probably custom lockfile), the developer can declare its own declaration file module.

Reply:
I personally do not need the package anymore since I'm currently using Yarn 2 which uses YAML (that, and I no longer have a need to parse the lockfile to begin with), so feel free to go ahead if you really think it's the right choice. 

The return type of `parse()` containing `LockFileObject` I feel is potentially dangerous since the original parse function is generic (and also reads YAML possibly to support v2 lockfiles): https://github.com/yarnpkg/yarn/blob/a75dbe36ca7f6d881d291a7231676d00d523bb39/src/lockfile/parse.js#L384-L431

The parameter type of `stringify()` is probably fine.